test/hotspot/jtreg/compiler/loopopts/LoopReductionHasControlOrBadInput.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @bug 8350576
+ * @summary Optimization bails out and hits an assert:
+ *          assert(false) failed: reduction has ctrl or bad vector_input
+ * @run main/othervm -XX:CompileCommand=compileonly,compiler.loopopts.LoopReductionHasControlOrBadInput::* -Xbatch -XX:-TieredCompilation compiler.loopopts.LoopReductionHasControlOrBadInput
```

Comment:
Learning from @eme64: You should probably add another `@run` without parameters to catch more things in the future.

Reply:
Thanks!